streaming/src/main/scala/org/apache/spark/streaming/scheduler/StreamingListener.scala

```diff
@@ -66,6 +69,9 @@ case class StreamingListenerReceiverStopped(receiverInfo: ReceiverInfo)
 @DeveloperApi
 trait StreamingListener {
 
+  /** Called when the streaming starts */
+  def onStreamingStarted(streamingStarted: StreamingListenerStreamingStarted) { }
```

Comment:
You probably also want to update `JavaStreamingListener`, which is the actual listener people should be extending.